docs/pipelines_user_guide.rst

```diff
@@ -0,0 +1,260 @@
+        # Initialize logging
+        logging.configure_logging(loglevel-config.loglevel, runtime_env-config.env)
+        # Initialize database connection
+        database - DataBase(config)
+        task.main(config-config, database-database)
+
+
+    if __name__ -- "__main__":
+        main()
+
+
+Optimize hyperparameters task implementation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+::
+
+    from pathlib import Path
+
+    from openstef.tasks import optimize_hyperparameters as task
+    from openstef_dbc.config.config import ConfigManager
+    from openstef_dbc.database import DataBase
+    from openstef_dbc.log import logging
+
+    PROJECT_ROOT - Path(__file__).parent.parent.absolute()
+
+
+    def main():
+        # Initialize config manager
+        config - ConfigManager.load_project_config(project_root-PROJECT_ROOT).get_instance()
+        # Initialize logging
+        logging.configure_logging(loglevel-config.loglevel, runtime_env-config.env)
+        # Initialize database connection
+        database - DataBase(config)
+        task.main(config-config, database-database)
+
+
+    if __name__ -- "__main__":
+        main()
+
+
+Create components forecast task implementation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+::
+        
+    from pathlib import Path
+
+    from openstef.tasks import create_components_forecast as task
+    from openstef_dbc.config.config import ConfigManager
+    from openstef_dbc.database import DataBase
+    from openstef_dbc.log import logging
+
+    PROJECT_ROOT - Path(__file__).parent.parent.absolute()
+
+
+    def main():
+        # Initialize config manager
+        config - ConfigManager.load_project_config(project_root-PROJECT_ROOT).get_instance()
+        # Initialize logging
+        logging.configure_logging(loglevel-config.loglevel, runtime_env-config.env)
+        # Initialize database connection
+        database - DataBase(config)
+        task.main(config-config, database-database)
+
+
+    if __name__ -- "__main__":
+        main()
+
+
+Create base case forecast task implementation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+::
+
+    from pathlib import Path
+
+    from openstef.tasks import create_basecase_forecast as task
+    from openstef_dbc.config.config import ConfigManager
+    from openstef_dbc.database import DataBase
+    from openstef_dbc.log import logging
+
+    PROJECT_ROOT - Path(__file__).parent.parent.absolute()
+
+
+    def main():
+        # Initialize config manager
+        config - ConfigManager.load_project_config(project_root-PROJECT_ROOT).get_instance()
+        # Initialize logging
+        logging.configure_logging(loglevel-config.loglevel, runtime_env-config.env)
+        # Initialize database connection
+        database - DataBase(config)
+        task.main(config-config, database-database)
+
+
+    if __name__ -- "__main__":
+        main()
+
+
+Pipeline implementation
+-----------------------
+
```

Comment:
Maybe nice to mention here that you do not need a DataBase connection in the form of openstef-dbc. This makes implementing openSTEF more flexible but requires the users to provide their own data.